Feature: Implement Kafka

CeNTePeDe opened the pull request.

CeNTePeDe committed:
removed list_streams
add constant variables
created:
- get_data_stream
Pydentic models:
- StreamerIn
- StreamerOut

CeNTePeDe committed:
created:
- router to streamer
- exception(InvalidIdInputError)
- created DAO

CeNTePeDe committed:
added
- id to url
- added Exception
removed const variables
removed url
removed:
- InvalidIdInputError

CeNTePeDe committed:
added
- added exception
- return None to update function

CeNTePeDe committed:
added
- streamer_routers.py
- streamer_models.py
- connection_streamers.py
- parse_streamer.py

CeNTePeDe committed:
override method update_streamer

created container:
- kafka
- zookeeper
created:
- producer
- consumer

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+        streamer_data = self.collection.find_one({"id": id})
+        if streamer_data is None:
+            return None
+        return StreamerIn(**streamer_data)
```
return StramerIN() if streamer data else none

slavadan commented on database/connection_streamers.py:
```diff
@@ -0,0 +1,58 @@
+        )
+        if streamer_update.modified_count == 0:
+            return None
+        return streamer_update.modified_count
```
возвращай просто streamer_update.modified_count

CeNTePeDe committed:
added
- update_one for create_item
- kafka_producer for streamer
- variables to .env file

slavadan commented on kafka_producers/kafka_streamers.py:
```diff
@@ -0,0 +1,19 @@
+def send_data_to_kafka_streamers(url):
+    from main import settings
+
+    producer = KafkaProducer(bootstrap_servers=settings.KAFKA_URL)
```
а вот это можно и в глобальную

slavadan commented on routers/streamer_routers.py:
```diff
@@ -0,0 +1,72 @@
+        auto_offset_reset=settings.AUTO_OFFSET_RESET,
+        bootstrap_servers=settings.KAFKA_URL,
+        consumer_timeout_ms=settings.CONSUMER_TIMEOUT_MS,
+    )
```
это тоже

CeNTePeDe committed:
add variable for kafka to constant_variables.py

CeNTePeDe committed:
create function:
- kafka_consumer_for_streamers
- kafka_consumer_for_products
